api/src/main/java/org/openmrs/module/metadatamapping/api/db/MetadataMappingDAO.java

```diff
@@ -130,4 +133,115 @@
+	 * @param metadataSetMemberId database id of the object
+	 * @return object or null, if does not exist
+	 */
+	MetadataSetMember getMetadataSetMember(Integer metadataSetMemberId);
+	
+	/**
+	 * Get members of a metadata set. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered 
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order 
+	 * will be unpredictable, if there are null sort weights in the set.
+	 * @param metadataSet metadata set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetMembers(String, String, int, int, RetiredHandlingMode)
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(MetadataSet metadataSet, int firstResult, int maxResults,
+	        RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get members of a metadata set. If members have {@link MetadataSetMember#getSortWeight()} set they will be ordered 
+	 * in ascending order according to said weight. Note that due to differences in database implementations, the order 
+	 * will be unpredictable, if there are null sort weights in the set.
+	 * @param metadataSourceName name of the source of the set
+	 * @param metadataSetCode code of the set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param retiredHandlingMode handle retired objects using this mode
+	 * @return list of members in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetMembers(MetadataSet, int, int, RetiredHandlingMode)
+	 */
+	List<MetadataSetMember> getMetadataSetMembers(String metadataSourceName, String metadataSetCode, int firstResult,
+	        int maxResults, RetiredHandlingMode retiredHandlingMode);
+	
+	/**
+	 * Get unretired metadata items in the set. If set members have {@link MetadataSetMember#getSortWeight()} set they will 
+	 * be ordered in ascending order according to said weight. Note that due to differences in database implementations, 
+	 * the order  will be unpredictable, if there are null sort weights in the set.
+	 * @param type type of the metadata items
+	 * @param metadataSet metadata set
+	 * @param firstResult zero based index of first result to get 
+	 * @param maxResults maximum number of results to get
+	 * @param <T> type of the metadata items
+	 * @return list of items in the order defined by the optional {@link MetadataSetMember#getSortWeight()} values
+	 * @see #getMetadataSetItems(Class, String, String, int, int)
+	 */
+	<T extends OpenmrsMetadata> List<T> getMetadataSetItems(Class<T> type, MetadataSet metadataSet, int firstResult,
```

Comment:
It is a bit tricky. It will not work if you want to retrieve all subtypes by specifying a super type, e.g. passing OpenmrsMetadata.class will return nothing, whereas you could expect it to return all items in that MetadataSet. It needs to be made clear in the javadoc that no subtypes will be returned. At the same time it would be nice to just call some method and get all OpenmrsMetadata regardless of type. We could have a method that does not take the type and returns List<? extends OpenmrsMetadata>.